framework/src/executioners/EigenExecutionerBase.C

```diff
@@ -197,7 +199,13 @@ EigenExecutionerBase::inversePowerIteration(unsigned int min_iter,
 
   // FIXME: currently power iteration use old and older solutions,
   // so save old and older solutions before they are changed by the power iteration
-  _eigen_sys.saveOldSolutions();
+  _problem.getNonlinearSystem().saveOldSolutions();
+  _problem.getAuxiliarySystem().saveOldSolutions();
```

Comment:
In good OOP design, you call methods on class instances without knowing the inside of the class. The class itself knows the best what to do. Not the caller.